src/Factory.php

```diff
@@ -2,152 +2,225 @@
-     * @param null|string $selector
+     * Create a new Validator instance.
      *
-     * @return \Proengsoft\JsValidation\Manager
+     * @param  array  $data
+     * @param  array  $rules
+     * @param  array  $messages
+     * @param  array  $customAttributes
+     * @return \Proengsoft\JsValidation\Validator
      */
-    public function make(array $rules, array $messages = array(), array $customAttributes = array(), $selector = null)
+    public function make(array $data, array $rules, array $messages = [], array $customAttributes = [])
     {
-        $validator = $this->validator->make([], $rules, $messages, $customAttributes);
+        $validator = $this->factory->make($data, $rules, $messages, $customAttributes);
+        //$delegated = new DelegatedValidator($validator);
+        /*
+        $delegated = $this->resolve($data, $rules, $messages, $customAttributes);
+        $validator = $delegated->getValidator();
+
+        if (! is_null($this->verifier)) {
+            $validator->setPresenceVerifier($this->verifier);
+        }
+
+        if (! is_null($this->container)) {
+            $validator->setContainer($this->container);
+        }
 
-        return $this->createValidator($validator, $selector);
+        $this->addExtensions($validator);
+        */
+        $jsValidator = $this->makeJsValidator($validator);
+        $this->configureJsRemote($jsValidator);
+
+        return $jsValidator;
+    }
+
+    protected function makeJsValidator(BaseValidator $validator)
+    {
+        $delegated = new DelegatedValidator($validator);
+        return new Validator($delegated);
     }
 
     /**
-     * Creates JsValidator instance based on FormRequest.
+     * Sets the session manager used to secure Ajax validations
      *
-     * @param $formRequest
-     * @param null $selector
+     * @param \Illuminate\Session\Store $store
+     */
+    public function setSessionStore(Store $store)
+    {
+        $this->sessionStore = $store;
+    }
+
+    /**
+     * Sets the session manager used to secure Ajax validations
      *
-     * @return Manager
+     * @return \Illuminate\Session\Store
+     */
+    public function getSessionStore()
+    {
+        return $this->sessionStore;
+    }
+
+    /**
+     * Enables or disable JsValidation Remote validations
      *
-     * @throws FormRequestArgumentException
+     * @param boolean $enabled
      */
-    public function formRequest($formRequest, $selector = null)
+    public function setJsRemoteEnabled($enabled)
     {
+        $this->jsRemoteEnabled = (bool)$enabled;
+    }
 
-        if (!is_subclass_of($formRequest, 'Illuminate\\Foundation\\Http\\FormRequest')) {
-            $className = is_object($formRequest) ? get_class($formRequest) : (string) $formRequest;
-            throw new FormRequestArgumentException($className);
-        }
+    /**
+     * Enables or disable JsValidation Remote validations
+     *
+     * @return bool
+     */
+    public function getJsRemoteEnabled()
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The 'getJsRemoteEnabled()' method which returns a boolean should be named 'is...()' or 'has...()'](https://www.codacy.com/app/albert-moreno-t/laravel-jsvalidation/file/993194591/issues/source?bid=2225045&fileBranchId=2869591#l128)